From eb2b8a2910562e0ea035bc979f7a1d321e6bb879 Mon Sep 17 00:00:00 2001
From: Mickey42302 <bgoodwin423@gmail.com>
Date: Thu, 28 May 2026 13:47:08 -0400
Subject: [PATCH] Always download the latest version of the modules


diff --git a/proxy/src/main/java/net/md_5/bungee/module/JenkinsModuleSource.java b/proxy/src/main/java/net/md_5/bungee/module/JenkinsModuleSource.java
index 8f5f34ae..b7434e7d 100644
--- a/proxy/src/main/java/net/md_5/bungee/module/JenkinsModuleSource.java
+++ b/proxy/src/main/java/net/md_5/bungee/module/JenkinsModuleSource.java
@@ -16,10 +16,10 @@ public class JenkinsModuleSource implements ModuleSource
     @Override
     public void retrieve(ModuleSpec module, ModuleVersion version)
     {
-        ProxyServer.getInstance().getLogger().info( "Attempting to Jenkins download module " + module.getName() + " v" + version.getBuild() );
+        ProxyServer.getInstance().getLogger().info( "Attempting to Jenkins download module " + module.getName() + " (latest build)" );
         try
         {
-            URL website = new URL( "https://hub.spigotmc.org/jenkins/job/BungeeCord/" + version.getBuild() + "/artifact/module/" + module.getName().replace( '_', '-' ) + "/target/" + module.getName() + ".jar" );
+            URL website = new URL( "https://hub.spigotmc.org/jenkins/job/BungeeCord/lastSuccessfulBuild/artifact/module/" + module.getName().replace( '_', '-' ) + "/target/" + module.getName() + ".jar" );
             URLConnection con = website.openConnection();
             // 15 second timeout at various stages
             con.setConnectTimeout( 15000 );
-- 
2.54.0

