From 6c3ad34b22ef982e429880b64de3a9193b68f62d Mon Sep 17 00:00:00 2001
From: Mickey42302 <bgoodwin423@gmail.com>
Date: Thu, 28 May 2026 14:04:08 -0400
Subject: [PATCH] Enable kick module by default


diff --git a/proxy/src/main/java/net/md_5/bungee/module/ModuleManager.java b/proxy/src/main/java/net/md_5/bungee/module/ModuleManager.java
index fec112e1..d94a646e 100644
--- a/proxy/src/main/java/net/md_5/bungee/module/ModuleManager.java
+++ b/proxy/src/main/java/net/md_5/bungee/module/ModuleManager.java
@@ -77,6 +77,7 @@ public class ModuleManager
             case 0:
                 defaults.add( "jenkins://cmd_alert" );
                 defaults.add( "jenkins://cmd_find" );
+                defaults.add( "jenkins://cmd_kick" );
                 defaults.add( "jenkins://cmd_list" );
                 defaults.add( "jenkins://cmd_send" );
                 defaults.add( "jenkins://cmd_server" );
-- 
2.54.0

