From 41915e491a7ee76280350b70ea887990bdec767f Mon Sep 17 00:00:00 2001
From: Mickey42302 <bgoodwin423@gmail.com>
Date: Thu, 28 May 2026 14:09:21 -0400
Subject: [PATCH] Add setRequestProperty


diff --git a/proxy/src/main/java/net/md_5/bungee/module/JenkinsModuleSource.java b/proxy/src/main/java/net/md_5/bungee/module/JenkinsModuleSource.java
index b7434e7d..36f44f3b 100644
--- a/proxy/src/main/java/net/md_5/bungee/module/JenkinsModuleSource.java
+++ b/proxy/src/main/java/net/md_5/bungee/module/JenkinsModuleSource.java
@@ -24,6 +24,7 @@ public class JenkinsModuleSource implements ModuleSource
             // 15 second timeout at various stages
             con.setConnectTimeout( 15000 );
             con.setReadTimeout( 15000 );
+            con.setRequestProperty( "User-Agent", "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/47.0.2526.106 Safari/537.36" );
 
             Files.write( ByteStreams.toByteArray( con.getInputStream() ), module.getFile() );
             ProxyServer.getInstance().getLogger().info( "Download complete" );
-- 
2.54.0

