From 99e794d775dc212c3ebbeecf31871e3b46efa58e Mon Sep 17 00:00:00 2001
From: Mickey42302 <bgoodwin423@gmail.com>
Date: Thu, 28 May 2026 14:14:54 -0400
Subject: [PATCH] Disable outdated build message (this is temporary)


diff --git a/proxy/src/main/java/net/md_5/bungee/BungeeCordLauncher.java b/proxy/src/main/java/net/md_5/bungee/BungeeCordLauncher.java
index 7acf942a..64115596 100644
--- a/proxy/src/main/java/net/md_5/bungee/BungeeCordLauncher.java
+++ b/proxy/src/main/java/net/md_5/bungee/BungeeCordLauncher.java
@@ -53,14 +53,15 @@ public class BungeeCordLauncher
 
             Calendar deadline = Calendar.getInstance();
             deadline.add( Calendar.WEEK_OF_YEAR, -8 );
-            if ( buildDate.before( deadline.getTime() ) )
-            {
-                System.err.println( "*** Warning, this build is outdated ***" );
-                System.err.println( "*** Please install the latest update as soon as possible ***" );
-                System.err.println( "*** You will not receive support for outdated builds ***" );
-                System.err.println( "*** Server will start in 10 seconds ***" );
-                Thread.sleep( TimeUnit.SECONDS.toMillis( 10 ) );
-            }
+            // This is temporary (no update server is available for Waterfall-CTD).
+            // if ( buildDate.before( deadline.getTime() ) )
+            // {
+            //     System.err.println( "*** Warning, this build is outdated ***" );
+            //     System.err.println( "*** Please install the latest update as soon as possible ***" );
+            //     System.err.println( "*** You will not receive support for outdated builds ***" );
+            //     System.err.println( "*** Server will start in 10 seconds ***" );
+            //     Thread.sleep( TimeUnit.SECONDS.toMillis( 10 ) );
+            // }
         }
 
         BungeeCord bungee = new BungeeCord();
-- 
2.54.0

