From f0ee0938f520437b1cf89a31bf71aa35c55d9fd9 Mon Sep 17 00:00:00 2001
From: Mickey42302 <bgoodwin423@gmail.com>
Date: Thu, 18 Jun 2026 19:09:07 -0400
Subject: [PATCH] Disable custom build warning (this is temporary)


diff --git a/proxy/src/main/java/net/md_5/bungee/module/ModuleManager.java b/proxy/src/main/java/net/md_5/bungee/module/ModuleManager.java
index 7d1d5d97..d643a9e5 100644
--- a/proxy/src/main/java/net/md_5/bungee/module/ModuleManager.java
+++ b/proxy/src/main/java/net/md_5/bungee/module/ModuleManager.java
@@ -37,7 +37,7 @@ public class ModuleManager
         ModuleVersion bungeeVersion = ModuleVersion.parse( proxy.getVersion() );
         if ( bungeeVersion == null )
         {
-            proxy.getLogger().warning( "Couldn't detect bungee version. Custom build?" );
+            // proxy.getLogger().warning( "Couldn't detect bungee version. Custom build?" );
             // return;
         }
 
-- 
2.55.0

